e2e/framework/testserver.go

```diff
@@ -417,11 +418,29 @@ func (t *TestServer) SendRawTx(
 	signer := crypto.NewEIP155Signer(100)
 	client := t.JSONRPC()
 
+	// if there is no from addr passed in we can find it in test server configuration
+	var blankAddr types.Address
+	if tx.From == blankAddr {
+		tx.From = t.Config.PremineAccts[0].Addr
+	}
```

Comment:
Why do we need this?

Reply:
Some tests were failing with not enough funds error message. 
This makes sure that that the sender always has enough funds, even if calling function doesn't specify From field. 

Should I investigate exactly which tests are failing when this check is omitted ?

Reply:
What tests fail when you omit this?

Reply:
I can't reproduce this fail anymore. Removed in 6138b95b4526bbbefa2f06a54581e344e2ebc9cd